sw/host/opentitanlib/src/transport/hyperdebug/mod.rs

```diff
@@ -404,29 +414,19 @@ impl<T: Flavor> Hyperdebug<T> {
         );
         let capabilities = u16::from_le_bytes([resp[2], resp[3]]);
         self.cmsis_google_capabilities.set(Some(capabilities));
-        self.inner
-            .usb_device
+        self.usb_device
             .borrow_mut()
             .release_interface(cmsis_interface.interface)?;
         Ok(capabilities)
     }
 }
 
-/// Internal state of the Hyperdebug struct, this struct is reference counted such that Gpio,
-/// Spi and Uart sub-structs can all refer to this shared data, which is guaranteed to live on,
-/// even if the caller lets the outer Hyperdebug struct run out of scope.
-pub struct Inner {
+#[derive(Clone)]
+pub struct CommandHandler {
```

Comment:
I think this CL would become a lot smaller, if we left the three fields `console_tty`, `usb_device` and `selected_spi` as only three members of an `Inner` struct, which would be reference counted.  This would be according to a convention that the `Inner` struct contains the all driver state that needs to be shared between any instances of SPI `Target`, `GpioPin`, or similar, which may out-live the main `Hyperdebug` struct.  It would be a shame to have to touch all the same places again, if we in the future need a fourth mutable field to be shared this way.  (Also, see my comment below on the "new" `Inner` struct.)